README.md

```diff
@@ -1,161 +1,87 @@
+**ES6 modules solve this problem.** Instead of importing the whole of `utils`, we can just import the `ajax` function we need:
 
 ```js
-// snippet from future-lodash.js
-export { partition } from './src/collection/partition';
-export { pluck } from './src/collection/pluck';
-export { reduce } from './src/collection/reduce';
-/* ...and so on... */
+import { ajax } from 'utils';
+
+var query = 'Rollup';
+ajax( 'https://api.example.com?search=' + query ).then( handleResponse );
 ```
 
-...and ES6 *bundlers* should handle importing in a much more granular fashion. Rather than importing an entire module, an intelligent bundler should be able to reason like so:
+Rollup statically analyses your code, and your dependencies, and includes the bare minimum in your bundle.
 
-* I need to import `pluck` from `future-lodash.js`
-* According to `future-lodash.js`, the definition of `pluck` can be found in `lodash/src/collection/pluck.js`
-* It seems that `pluck` depends on `map` and `property`, which are in... *these* files
-* ...
-* Right, I've found all the function definitions I need. I can just include those in my bundle and disregard the rest
 
-In other words, the 'tree-shaking' approach of throwing everything in then removing the bits you don't need is all wrong - instead, we should be selective about what we include in the first place.
+## Shouldn't we be writing those utilities as small modules anyway?
 
-This is not a trivial task. There are almost certainly a great many complex edge cases. Perhaps it's not possible. But I intend to find out.
+[Not always, no.](https://medium.com/@Rich_Harris/small-modules-it-s-not-quite-that-simple-3ca532d65de4)
 
 
-## Goals
+## Don't minifiers already do this?
 
-* Maximally efficient bundling
-* Ease of use
-* Flexible output - CommonJS, AMD, UMD, globals, ES6, System etc
-* Speed
-* Character-accurate sourcemaps
-* Eventually, port the functionality to esperanto
+If you minify code with something like [UglifyJS](https://github.com/mishoo/UglifyJS2) (and you should!) then some unused code will be removed:
 
-### Secondary goals
+```js
+(function () {
+  function foo () {
+    console.log( 'this function was included!' );
+  }
+
+  function bar () {
+    console.log( 'this function was not' );
+    baz();
+  }
+
+  function baz () {
+    console.log( 'neither was this' );
+  }
+
+  foo();
+})();
+```
 
-* Support for legacy module formats
-* Respect for original formatting and code comments
+A minifier can detect that `foo` gets called, but that `bar` doesn't. When we remove `bar`, it turns out that we can also remove `baz`.
 
+Unfortunately, **traditional modules – CommonJS and AMD – make this kind of optimisation next to impossible**. Rather than *excluding dead code*, we should be *including live code*. That's only possible with ES6 modules.
```

Comment:
You and I know why traditional modules make this next to impossible, but the reader of this may not. Should we explain or link to an explanation? It probably wouldn't take more than a sentence or two to explain.